src/Impostor.Server/Net/State/Game.Data.cs

```diff
@@ -381,6 +382,11 @@ private async ValueTask OnDestroyAsync(InnerNetObject netObj)
 
                 case InnerPlayerControl control:
                 {
+                    if (GameState != GameStates.Started)
+                    {
+                        GameNet.GameData?.RemovePlayer(control);
```

Comment:
I don't see yet how this code is related to the main subject of this PR

Reply:
Pretty sure it caused problems with player state array sorting, @Mr-VIT

Reply:
@js6pak so does this resolve the issue where we had to sort by netid to get the proper player order?

Reply:
this is necessary to match `GameData._allPlayers` with host.